TalZivelin/TZivelin-HW-W01.js

```diff
@@ -0,0 +1,125 @@
+
+b) Write a series of expressions to adjust each of x, y, and z halfway toward their mean. That is, reset the value of each variable to something based on its previous value. 
+
+Use the mean variable from part a, 
+x = x + mean / 2; 
+y =y + mean / 2;
+z = z + mean / 2;
+	
+
+3) (Difficulty: easyish)
+
+Suppose you're encoding geometric shapes in a Cartesian (2D) coordinate system, and you represent a rectangle with four numeric variables:
+
+    l : horizontal position of left edge;
+    r : horizontal position of right edge;
+    t : vertical position of top edge;
+    b : vertical position of bottom edge.
+
+a) Write an expression for the rectangle's area. - a = (r - l) * (t - b)
+
+b) Write an expression which is true if the rectangle is taller than it is wide, and false otherwise. - (t - b) > (r - l)
+
+c) Write an expression for the circumference of the biggest circle which can fit inside the rectangle. (Hint: you'll need logic similar to that in b.) - 
+
+var width = (r - l);
+var height = (t - b);
+var diameter;
+ 
+if (width > height) 
+	diameter = Math.PI height;
+		else
+  	diameter = Math.PI width;
+
+
+d) Write an expression for the area of the smallest circle which completely encloses (i.e. circumscribes) the rectangle.
+
+r2 = width/2 + height/2
+
+e) Imagine subdividing your rectangle into 3 equal rows and 3 equal columns, which would create 9 smaller rectangles, identical in shape but varying by position. Define four new variables describing the centermost small rectangle. (Hint: one of the many solutions is very similar to the solution of 2b above.)
+
+4) (Difficulty: easyish)
+
+Imagine that the squares of an ordinary checkerboard are numbered in two different ways:
+
+    Each square has integer coordinates (R,C) describing its row and column. Both values should be in the range 0..7, so that the upper-left square is at (0,0) and the bottom-right is at (7,7).
+
+    Each square has a unique integer number N from 0 to 63. These numbers run sequentially left-to-right one row at a time, top to bottom. Therefore the upper-left square has N==0 and the bottom-right has N==63.
+
+a) Given a particular R and C, find the corresponding N. That is, write an expression for variable N containing variables R and C.
+	N = R * 8 + C
+
+b) Given N, find R. Write an expression for R which contains N.
+	R = Math.floor(N / 8);
+
+c) Given N, find C. Write an expression for C which contains N.
+	C = N % 8. Since there are 8 columns mod by 8 will tell you which column you're on.
+
+d) Assume the squares are colored black and white, with the upper-left square black. Write an expression to set a variable color to either 'black' or 'white', describing the square identified by variables R,C, and N. (Hint: If you prefer, you may use a pair of conditional statements instead of a single expression.) -  
+
+if (N % 2 = 0)	{
+	square = black
+}
+	else {
+	square = white
+}
```

Comment:
4d) The tricky bit here is that checkerboards alternate both vertically and horizontally. This equation only alternates horizontally; you'd be left with stripes, basically.